crypto/schemes.go

```diff
@@ -197,7 +194,46 @@ func NewPedersenBLSUnchainedSwapped() (cs *Scheme) {
 		ThresholdScheme: ThresholdScheme,
 		AuthScheme:      AuthScheme,
 		DKGAuthScheme:   DKGAuthScheme,
-		Pairing:         Pairing,
+		IdentityHash:    IdentityHashFunc,
+		DigestBeacon:    DigestFunc,
+	}
+}
+
+// SigsOnG1ID is the scheme id used to set unchained randomness on beacons with signatures on G1 that are
+// compliant with the hash to curve RFC.
+const SigsOnG1ID = "bls-unchained-g1-rfc"
+
+// NewPedersenBLSUnchainedG1 instantiate a scheme of type "bls-unchained-on-g1" which is also unchained, only
+// hashing the round number as the message being signed in beacons. This scheme is also compatible with
+// "timelock encryption" as done by tlock.
+// This schemes has the group public key on G2, so 96 bytes, and the beacon signatures on G1, so 48 bytes.
+// This means databases of beacons produced with this scheme are almost half the size of the other schemes.
+//
+//nolint:dupl
+func NewPedersenBLSUnchainedG1() (cs *Scheme) {
+	var Pairing = bls.NewBLS12381Suite()
+	var KeyGroup = Pairing.G2()
+	var SigGroup = Pairing.G1()
+	// using G1 for the ThresholdScheme since it allows beacons to have shorter signatures, reducing the size of any
+	// database storing all existing beacons by half compared to using G2.
+	var ThresholdScheme = tbls.NewThresholdSchemeOnG1(Pairing)
+	var AuthScheme = signBls.NewSchemeOnG1(Pairing)
+	var DKGAuthScheme = schnorr.NewScheme(&schnorrSuite{KeyGroup})
+	var IdentityHashFunc = func() hash.Hash { h, _ := blake2b.New256(nil); return h }
+	// Unchained means we're only hashing the round number
+	var DigestFunc = func(b hashableBeacon) []byte {
```

Comment:
Second time we're using this function, so now it's time to put it as its own function so it's factored out and clear it's the "unchained" hashing we're using ?